File: FINALGAMECAPSTONE/Assets/SCRIPTS/coin.cs
 using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class coin : MonoBehaviour {
	void Start () {
	}
	void OnTriggerEnter2D (Collider2D other){

		Destroy(gameObject);
	
		Score.diamonds++;
		Debug.Log (Score.diamonds);
	}
}

File: FINALGAMECAPSTONE/Assets/SCRIPTS/Coin.js
#pragma strict

private var score : Score;

function Start () {

score = GameObject.FindGameObjectWithTag("Player").GetComponent(Score);
}

function OnTriggerEnter2D (other : Collider2D){

if (other.tag == "Player"){
score.diamonds += 1;
Destroy(gameObject);
}
}